### PR TITLE
engine: build and deploy from existing

### PR DESCRIPTION
**Pull request by jazzdan**
I haven't tested this end-to-end yet but wanted to get your eyes on it

### CODE REVIEW

File: internal/service/manager.go [hunk 1]
```diff
@@ -76,3 +77,14 @@ func (m *memoryManager) Remove(n model.ServiceName) {
 
 	delete(m.services, n)
 }
+
+func (m *memoryManager) Get(n model.ServiceName) (model.Service, error) {
```
Comment (jazzdan) [inline internal/service/manager.go]: I should add a super simple test for this.

File: internal/engine/upper.go [hunk 2]
```diff
@@ -2,20 +2,22 @@ package engine
...
 
-func UpService(ctx context.Context, service model.Service, watch bool, stdout io.Writer, stderr io.Writer) error {
-	bad, err := NewLocalBuildAndDeployer()
+func UpService(ctx context.Context, service model.Service, watch bool, stdout io.Writer, stderr io.Writer, manager service.Manager) error {
+	bad, err := NewLocalBuildAndDeployer(manager)
 	if err != nil {
 		return err
 	}
-	buildToken, err := bad.BuildAndDeploy(ctx, service, nil)
+	_, err = bad.BuildAndDeploy(ctx, service, nil)
```
Comment (jazzdan) [inline internal/engine/upper.go]: future of build token is uncertain in my head in this world

File: internal/proto/server_adaptor.go [hunk 3]
```diff
@@ -27,7 +27,7 @@ func (s *grpcServer) CreateService(req *CreateServiceRequest, d Daemon_CreateSer
 	outputStream := MakeStdoutStderrWriter(sendOutput)
 
 	service := serviceP2D(req.Service)
-	err := engine.UpService(d.Context(), service, req.Watch, outputStream.stdout, outputStream.stderr)
+	err := engine.UpService(d.Context(), service, req.Watch, outputStream.stdout, outputStream.stderr, s.sm)
```
Comment (jazzdan) [inline internal/proto/server_adaptor.go]: just doin' some DI

File: internal/engine/upper.go [hunk 4]
```diff
@@ -2,20 +2,22 @@ package engine
...
 	"io"
+
+	"github.com/windmilleng/tilt/internal/model"
+	"github.com/windmilleng/tilt/internal/service"
 )
 
-func UpService(ctx context.Context, service model.Service, watch bool, stdout io.Writer, stderr io.Writer) error {
-	bad, err := NewLocalBuildAndDeployer()
+func UpService(ctx context.Context, service model.Service, watch bool, stdout io.Writer, stderr io.Writer, manager service.Manager) error {
+	bad, err := NewLocalBuildAndDeployer(manager)
```
Comment (maiamcc) [inline internal/engine/upper.go]: i'm probably disproportionately confused by calling this `bad` but... yeah.

File: internal/engine/build_and_deployer.go [hunk 5]
```diff
@@ -3,30 +3,34 @@ package engine
...
+	"github.com/windmilleng/tilt/internal/service"
 	"github.com/windmilleng/wmclient/pkg/dirs"
-	"os"
 )
 
 type BuildToken interface{}
 
 type BuildAndDeployer interface {
 	BuildAndDeploy(ctx context.Context, service model.Service, token BuildToken) (BuildToken, error)
+	BuildAndDeployFromExisting(ctx context.Context, name model.ServiceName) error
```
[the remark this answers is not in the document]
Comment (landism) [reply internal/engine/build_and_deployer.go]: (which this PR made me realize means we'd need a list of filenames on `BuildAndDeploy`)

**Comment by jazzdan** [comment]
@landism @nicks here's a new version that does what we talked about. I'll leave the manager code in for now since it seems like we agree that we want to be keeping track of that stuff eventually, even if we aren't using it now.

**Pull request merged**